src/adapters/SDKAdapter.js

```diff
@@ -1,24 +1,22 @@
 import Adapter from './Adapter';
+import {Observable} from 'rxjs';
 
 import sdk from '../data/sdk';
 
 export default class SDKAdapter extends Adapter {
   getPerson(id) {
-    return new Promise((resolve) => {
-      // Subscribe and wait for our reducer
-      sdk.store.subscribe(async () => {
-        const person = await sdk.store.getState();
-        resolve(person);
-      });
-
-      // Dispatch the fetch action
-      sdk.store.dispatch({
-        type: 'PERSON_DETAILS',
-        payload: {
-          id: id
-        }
-      });
-    });
+    sdk.getPerson(id);
```

Comment:
Don't we need to subscribe before doing the `sdk.getPerson`? That way if it immediately returns, we don't miss the subscription update?